### PR TITLE
net: replace reth-primitive imports with alloy-eips

### PR DESCRIPTION
**Pull request by estensen**
Progress on https://github.com/paradigmxyz/reth/issues/10749

### CODE REVIEW

**Comment by estensen** [comment]
Removing this unused import leads to this

```
error[E0432]: unresolved import `alloy_eips`
  --> crates/net/eth-wire/tests/fuzz_roundtrip.rs:51:9
   |
51 |     use alloy_eips::BlockHashOrNumber;
   |         ^^^^^^^^^^ use of undeclared crate or module `alloy_eips`
   |
help: there is a crate or module with a similar name
   |
51 |     use alloy_rlp::BlockHashOrNumber;
   |         ~~~~~~~~~
```

File: crates/net/p2p/src/test_utils/headers.rs [hunk 1]
```diff
@@ -108,7 +108,7 @@ impl TestDownload {
             let request = HeadersRequest {
                 limit: self.limit,
                 direction: HeadersDirection::Rising,
-                start: reth_primitives::BlockHashOrNumber::Number(0), // ignored
+                start: alloy_eips::BlockHashOrNumber::Number(0), // ignored
```
Comment (mattsse) [inline crates/net/p2p/src/test_utils/headers.rs:111]: you can probably use 0u64.into() here

**Comment by onbjerg** [comment]
> Removing this unused import leads to this
> 
> ```
> error[E0432]: unresolved import `alloy_eips`
>   --> crates/net/eth-wire/tests/fuzz_roundtrip.rs:51:9
>    |
> 51 |     use alloy_eips::BlockHashOrNumber;
>    |         ^^^^^^^^^^ use of undeclared crate or module `alloy_eips`
>    |
> help: there is a crate or module with a similar name
>    |
> 51 |     use alloy_rlp::BlockHashOrNumber;
>    |         ~~~~~~~~~
> ```

you need to demote it to a dev dependency instead, it's used in a test

**Review by onbjerg — APPROVED** [review]
lgtm otherwise

**Review by mattsse — APPROVED** [review]

**Pull request merged**